test/controllers/testOnly/ControlDecisionControllerSpec.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.testOnly
+
+import base.SpecBase
+import generators.Generators
+import matchers.JsonMatchers
+import models.LocalReferenceNumber
+import models.departure.ControlDecision
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatestplus.mockito.MockitoSugar
+import play.api.inject.bind
+import play.api.libs.json.{JsObject, Json}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.twirl.api.Html
+import services.DepartureMessageService
+
+import scala.concurrent.Future
+
+class ControlDecisionControllerSpec extends SpecBase with MockitoSugar with JsonMatchers with Generators {
+
+  private val mockDepartureMessageService = mock[DepartureMessageService]
+
+  override def beforeEach: Unit = {
+    reset(mockDepartureMessageService)
+    super.beforeEach
+  }
+
+  "ControlDecision Controller" - {
```

Comment:
We should add the test for the error handing code. I suggested changing the behaviour to be a 500 with Technical Difficulties page